resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/MultiValuedAbstractParamConverter.java

```diff
@@ -0,0 +1,107 @@
+package org.jboss.resteasy.plugins.providers;
+
+
+import java.lang.reflect.Method;
+import java.util.Collection;
+
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.RuntimeDelegate.HeaderDelegate;
+
+import org.jboss.resteasy.core.StringParameterInjector;
+import org.jboss.resteasy.resteasy_jaxrs.i18n.Messages;
+
+/**
+ * @author Marek Kopecky mkopecky@redhat.com
+ * @author Ron Sigal rsigal@redhat.com
+ */
+public abstract class MultiValuedAbstractParamConverter
+{
+   protected StringParameterInjector stringParameterInjector;
+   @SuppressWarnings("rawtypes")
+   protected ParamConverter paramConverter;
+   @SuppressWarnings("rawtypes")
+   protected HeaderDelegate headerDelegate;
+   protected String separator;
+
+   public MultiValuedAbstractParamConverter(StringParameterInjector stringParameterInjector, String separator)
+   {
+      this.stringParameterInjector = stringParameterInjector;
+      this.paramConverter = stringParameterInjector.getParamConverter();
+      this.headerDelegate = stringParameterInjector.getHeaderDelegate();
+      this.separator = separator;
+   }
+
+   ////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+   @SuppressWarnings({ "unchecked" })
+   protected String stringify(Collection<?> value)
+   {
+      Method saveAsString = null;
+      if (paramConverter == null && headerDelegate == null)
+      {
+         try
+         {
+            Object o = value.iterator().next();
+            saveAsString = o.getClass().getMethod("saveAsString");
```

Comment:
What about to add description of `saveAsString` method to docs? (StringConverter.xml)

Reply:
@marekkopecky, I eliminated saveAsString(). I just look for toString() now.